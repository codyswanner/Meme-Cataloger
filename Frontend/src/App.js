import React, { useState } from 'react';
import { Box, createTheme, CssBaseline, Toolbar } from '@mui/material';
import { ThemeProvider } from '@emotion/react';

import TopAppBar from './TopAppBar/TopAppBar';
import LeftDrawer from './LeftDrawer/LeftDrawer';
import VirtuosoGridWrapper from './VirtualizedImageList';
import useFilterSocket from './SupportingModules/useFilterSocket';
import AppDataContext from './SupportingModules/AppDataContext';
import DropHandler from './DropHandler';
import UploadFilesContextProvider from './SupportingModules/UploadFilesContext';


export function AppChildren() {
  return (
      <Box sx={{ display: 'flex' }}>
          {/* Provided by MUI Material for basic styling */}
          <CssBaseline />

          {/* Navigation banner/AppBar at top of the page */}
          <TopAppBar/>

          {/* Drawer on left side, holds tag filter options */}
          <LeftDrawer/>

          {/* Main content area */}
          <Box component="main" sx={{ flexGrow: 1 }}>
              <Toolbar />{/* Hides under AppBar, push below to open space */}
              <div className='flex-container'>
                  <VirtuosoGridWrapper/>
              </div>
          </Box>
      </Box>
  );

}

/**
 * Root component that renders all other components.
 * Activates useFilterSocket hook,
 * which is the hub for WebSocket communications with backend.
 * 
 * @param {object} props Contains props passed into the component.
 * @param {Array} props.apiData Data for initial page load.
 * 
 * @returns The root App component to render all other components.
 */
function App(props) {

  // useFilterSocket receives messages for updates to appData.
  // appData contains Image, Tag and ImageTag data.
  const [drawerOpen, setDrawerOpen] = useState(false);
  const [isClosingDrawer, setIsClosingDrawer] = useState(false);
  const [editTags, setEditTags] = useState(false);
  const [uploadDialogOpen, setUploadDialogOpen] = useState(false);
  const appData = useFilterSocket(props.apiData);
  const appState = {
    drawerOpen: drawerOpen,
    setDrawerOpen: setDrawerOpen,
    isClosingDrawer: isClosingDrawer,
    setIsClosingDrawer: setIsClosingDrawer,
    editTags: editTags,
    setEditTags: setEditTags,
    uploadDialogOpen: uploadDialogOpen,
    setUploadDialogOpen: setUploadDialogOpen
  };
  // createTheme sets background color
  const theme = createTheme({
    palette: {
      background: {
        default: '#666666',
      },
    },
    components: {
      MuiToolbar: {
        styleOverrides: {
          regular: {
            // Toolbar is used to push content out from under the AppBar.
            // AppBar is larger than standard size, because of IconButton for the drawer.
            // So, increase the max Toolbar size to allow tall enough Toolbars for this.
            maxHeight: 1000
          }
        }
      }
    }
  });



  return (
    <>
    {/* https://mui.com/material-ui/customization/theming/#themeprovider */}
    <ThemeProvider theme={theme}>
      {/* https://react.dev/reference/react/useContext */}
      <AppDataContext.Provider value={{appData, appState}}>
      <UploadFilesContextProvider>
        <DropHandler>
          {props.children}
        </DropHandler>
      </UploadFilesContextProvider>
      </AppDataContext.Provider>
    </ThemeProvider>
    </>
  );
}

export default App;
