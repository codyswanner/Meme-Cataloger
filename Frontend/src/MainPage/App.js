import React, { useState } from 'react';
import { Box, createTheme, CssBaseline, Toolbar } from '@mui/material';
import { ThemeProvider } from '@emotion/react';

import ApiCall from '../SupportingModules/ApiCall';
import TopAppBar from './TopAppBar';
import TagDrawer from './TagDrawer/TagDrawer';
import VirtuosoGridWrapper from './ImageList/VirtualizedImageList';
import useFilterSocket from '../SupportingModules/useFilterSocket';
import AppDataContext from '../SupportingModules/AppDataContext';
import DropHandler from './Uploads/DropHandler';



// initial data for page load, provided by Django API.
let apiData = await ApiCall();

/**
 * Root component that renders all other components.
 * Activates useFilterSocket hook,
 * which is the hub for WebSocket communications with backend.
 * 
 * @param {object} props Contains props passed into the component.
 * 
 * @returns The root App component to render all other components.
 */
export default function App() {

  // useFilterSocket receives messages for updates to appData.
  // appData contains Image, Tag and ImageTag data.
  const [drawerOpen, setDrawerOpen] = useState(false);
  const [isClosingDrawer, setIsClosingDrawer] = useState(false);
  const [editTags, setEditTags] = useState(false);
  const [files, setFiles] = useState([]);
  const [uploadDialogOpen, setUploadDialogOpen] = useState(false);
  const appData = useFilterSocket(apiData);
  const appState = {
    drawerOpen: drawerOpen,
    setDrawerOpen: setDrawerOpen,
    isClosingDrawer: isClosingDrawer,
    setIsClosingDrawer: setIsClosingDrawer,
    editTags: editTags,
    setEditTags: setEditTags,
    files: files,
    setFiles: setFiles,
    uploadDialogOpen: uploadDialogOpen,
    setUploadDialogOpen: setUploadDialogOpen
  };
  // createTheme sets background color
  const theme = createTheme({
    palette: {
      background: {
        default: '#666666',
      },
    },
    components: {
      MuiToolbar: {
        styleOverrides: {
          regular: {
            // Toolbar is used to push content out from under the AppBar.
            // AppBar is larger than standard size, because of IconButton for the drawer.
            // So, increase the max Toolbar size to allow tall enough Toolbars for this.
            maxHeight: 1000
          }
        }
      }
    }
  });

  return (
    <>
    {/* https://mui.com/material-ui/customization/theming/#themeprovider */}
    <ThemeProvider theme={theme}>
      {/* https://react.dev/reference/react/useContext */}
      <AppDataContext.Provider value={{appData, appState}}>
        <DropHandler>
          <Box sx={{ display: 'flex' }}>
            {/* Provided by MUI Material for basic styling */}
            <CssBaseline />

            {/* Navigation banner/AppBar at top of the page */}
            <TopAppBar/>

            {/* Drawer on left side, holds tag filter options */}
            <TagDrawer/>

            {/* Main content area */}
            <Box component="main" sx={{ flexGrow: 1 }}>
                <Toolbar />{/* Hides under AppBar, push below to open space */}
                <div className='flex-container'>
                    <VirtuosoGridWrapper/>
                </div>
            </Box>
          </Box>
        </DropHandler>
      </AppDataContext.Provider>
    </ThemeProvider>
    </>
  );
};
