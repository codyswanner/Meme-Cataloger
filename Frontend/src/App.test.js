import renderer from 'react-test-renderer';
import App from './App';


describe('App', () => {
    test('snapshot', () => {
        // Will not actually be checked in this test
        const mockApiData = [
            [], [], []
        ]
        
        const tree = renderer
            .create(<App apiData={mockApiData}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
