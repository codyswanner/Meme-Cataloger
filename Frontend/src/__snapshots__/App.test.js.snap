// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="MuiBox-root css-k008qs"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-ivnu1y-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-1961ak2-MuiTypography-root"
      >
        Meme-opolis
      </div>
      <label
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1y3seza-MuiButtonBase-root-MuiButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="UploadIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 20h14v-2H5v2zm0-10h4v6h6v-6h4l-7-7-7 7z"
            />
          </svg>
        </span>
        Upload
      </label>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked css-1fnbm7f-MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      />
       
      <div
        className="MuiBox-root css-1v8x7dw"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
           
        </ul>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
           
          <div
            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Archive
              </span>
            </div>
          </div>
          <div
            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Trash
              </span>
            </div>
          </div>
        </ul>
      </div>
    </div>
  </div>
  <main
    className="MuiBox-root css-i9gxme"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    />
     
    <div
      className="flex-container"
    >
      <ul
        className="MuiImageList-root MuiImageList-masonry css-10riir9-MuiImageList-root"
        style={
          Object {
            "columnCount": 4,
            "columnGap": 4,
          }
        }
      >
         
      </ul>
    </div>
  </main>
</div>
`;
